File: gems/decomplex/examples/rust/temporal-ordering-pressure.rs
pub struct TemporalOrderExample {
    a: i32,
    b: i32,
}

impl TemporalOrderExample {
    pub fn one(&mut self) {
        self.a = 1;
    }

    pub fn two(&mut self) {
        self.a = 2;
        self.b = 3;
    }

    pub fn three(&mut self) {
        self.b = 4;
    }

    pub fn reader(&self) -> i32 {
        self.a
    }
}

File: gems/decomplex/examples/go/temporal-ordering-pressure.go
package main
type TemporalOrderExample struct { a int; b int }
func (self *TemporalOrderExample) One() { self.a = 1 }
func (self *TemporalOrderExample) Two() { self.a = 2; self.b = 3 }
func (self *TemporalOrderExample) Three() { self.b = 4 }
func (self *TemporalOrderExample) Reader() int { return self.a }
